s3_storage_client_test.go

```diff
@@ -793,3 +793,47 @@ func TestConvertContentLength(t *testing.T) {
 		})
 	}
 }
+
+func TestGetS3Endpoint(t *testing.T) {
+	testcases := []struct {
+		desc string
+		in   execResponseStageInfo
+		out  string
+	}{
+		{
+			desc: "when useS3RegionalURL and is enabled and domain starts with cn",
+			in: execResponseStageInfo{
+				UseS3RegionalURL: true,
```

Comment:
let's add also cases with different values of `UseRegionalURL` and with empty endpoint and region not starting with `cn-`

Reply:
I added testing cases.